### PR TITLE
Add signatures for elementwise pow operator and matrix power

### PR DESCRIPTION
**Pull request by andrjohns**
This PR adds the signatures and tests for the new elementwise ```.^``` operator, as well as for the ```matrix_power``` function.

There's a good chance that I haven't taken the most efficient route for adding the ```.^``` operator, as well as defining the signatures, so let me know if there's a better way that I should be doing this.

### CODE REVIEW

**Comment by andrjohns** [comment]
Good call, opened a docs PR now. I ran those dune commands and got ```nothing to promote```. As for the test failures on jenkins, I'm also seeing that on the current ```master``` branch so I'm not sure where that's coming from

**Comment by rok-cesnovar** [comment]
Oh then you need to merge current master in and run tests again.

**Comment by andrjohns** [comment]
Yep, that did it.

File: test/integration/signatures/stan_math_sigs.expected [hunk 1]
```diff
@@ -14142,7 +14143,60 @@ int[] poisson_rng(row_vector)
...
+real[,] pow(int, int[,])
+real[,] pow(int, real[,])
+real[,] pow(real, int[,])
+real[,] pow(real, real[,])
+real[,] pow(int[,], int)
+real[,] pow(int[,], real)
+real[,] pow(int[,], int[,])
+real[,] pow(real[,], int)
+real[,] pow(real[,], real)
+real[,] pow(real[,], real[,])
```
Comment (rok-cesnovar) [inline test/integration/signatures/stan_math_sigs.expected:14199]: Do we not want

```
vector[,] pow(int, vector[,])
vector[,] pow(real, vector[,])
vector[,] pow(vector[,], int)
vector[,] pow(vector[,], real)
vector[,] pow(vector[,], vector[,])
vector[,,] pow(int, vector[,,])
vector[,,] pow(real, vector[,,])
vector[,,] pow(vector[,,], int)
vector[,,] pow(vector[,,], real)
vector[,,] pow(vector[,,], vector[,,])
```

Comment (andrjohns) [reply test/integration/signatures/stan_math_sigs.expected:14199]: We can nest as deeply as we like, I just wasn't sure what depth we'd want to expose. Should I go up to ```vector[,,]```?

Comment (rok-cesnovar) [reply test/integration/signatures/stan_math_sigs.expected:14199]: With unary functions that support arbitrary nesting with array we tend to go to 7 dimensions.

Comment (andrjohns) [reply test/integration/signatures/stan_math_sigs.expected:14199]: I'm happy to keep the binary vectorisation consistent with the unary vectorisation, I'll add in those signatures and tests

Comment (andrjohns) [reply test/integration/signatures/stan_math_sigs.expected:14199]: Quick question, how deep should the tests go? It looks like the ```exp``` tests only test up to [three levels of nesting](https://github.com/stan-dev/stanc3/blob/master/test/integration/good/function-signatures/math/matrix/exp.stan)

Comment (rok-cesnovar) [reply test/integration/signatures/stan_math_sigs.expected:14199]: Three is fine probably. Things won't change much after that.

**Comment by andrjohns** [comment]
Ready for re-review now

**Comment by rok-cesnovar** [comment]
Merging this as the docs are ready, just need a review.

**Pull request merged**